addons/website_slides/views/website_slides_templates_course.xml

```diff
@@ -594,23 +600,24 @@
 <template id='lesson_card' name="Lesson Card">
     <div class="card w-100 o_wslides_lesson_card mb-4">
         <t t-set="can_access" t-value="slide.is_preview or channel.is_member or channel.can_publish"/>
+        <t t-set="search_param" t-value="'search_category=%s' % search_category.id if search_category else 'search_tag=%s' % search_tag.id if search_tag else 'search_slide_type=%s' % search_slide_type if search_slide_type else 'search_uncatgorized=1' if search_uncategorized else ''"/>
 
         <t t-if="slide.image">
             <t t-set="lesson_image" t-value="'/web/image/slide.slide/%s/image' % slide.id"/>
-            <a t-if="can_access" t-attf-href="/slides/slide/#{slug(slide)}" t-title="slide.name">
+            <a t-if="can_access" t-attf-href="/slides/slide/#{slug(slide)}?#{search_param}" t-title="slide.name">
```

Comment:
Maybe try

`#{'?' if search_param else ''}#{search_param}`

Or put the '?' in the search_param variable and rename it `query_string` or something.